### PR TITLE
[14.0][FIX] sale_blanket_order and sale_order_revision: add sudo

### PR DESCRIPTION
**Pull request by csil300**
Add sudo() to sale_blanket_order and sale_order_revision modules.

Steps to reproduce issue:

Sales > Orders > Orders > Select a Sales Order with a Revision > Click Prev. revisions smart button
Sales > Orders > Blanket Orders > Select a Blanket Order > Click Sale Orders smart button
Sales > Orders > Blanket Orders > Select a Blanket Order > Click Lines smart button

### CODE REVIEW

**Comment by dreispt** [comment]
@csil300 Can you squash the commits please?
You can avoid that that by using `git commit --amend` to correct the commit, instead of adding a new one.

**Comment by dreispt** [comment]
Also, you need a separate commit for each module, otherwise version migration will be a problem.
I suggest you to  `git reset 14.0`, and then add and commit each module changes,.

**Review by dreispt — CHANGES_REQUESTED** [review]
Please split separate modules changes in separate commits.

**Review by dreispt — APPROVED** [review]

File: sale_blanket_order/models/blanket_orders.py [hunk 1]
```diff
@@ -327,7 +327,7 @@ def action_cancel(self):
 
     def action_view_sale_orders(self):
         sale_orders = self._get_sale_orders()
-        action = self.env.ref("sale.action_orders").read()[0]
+        action = self.env.ref("sale.action_orders").sudo().read()[0]
```
Comment (rousseldenis) [inline sale_blanket_order/models/blanket_orders.py:330]: You should use _for_xml_id():

```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id("sale.action_orders")
```

File: sale_blanket_order/models/blanket_orders.py [hunk 2]
```diff
@@ -336,9 +336,13 @@ def action_view_sale_orders(self):
         return action
 
     def action_view_sale_blanket_order_line(self):
-        action = self.env.ref(
-            "sale_blanket_order" ".act_open_sale_blanket_order_lines_view_tree"
-        ).read()[0]
+        action = (
```
Comment (rousseldenis) [inline sale_blanket_order/models/blanket_orders.py:339]: Same as above

File: sale_order_revision/models/sale_order.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def _prepare_revision_data(self, new_revision):
     def action_view_revisions(self):
         self.ensure_one()
         action = self.env.ref("sale.action_orders")
-        result = action.read()[0]
+        result = action.sudo().read()[0]
```
Comment (rousseldenis) [inline sale_order_revision/models/sale_order.py:38]: Same as above

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@csil300

**Comment by rousseldenis** [comment]
Closing as @csil300 has no more activity

**Pull request closed without merging**